app/models/membership.rb

```diff
@@ -25,20 +25,17 @@ def self.joins_for
     @parent
   end
 
-
   def self.scope_for(action, user, opts={})
     return all if user.is_admin?
     roles, _ = parent_class.roles(action)
     accessible_groups = user.user_groups.where.overlap(memberships: {roles: roles})
-    query = where(user_group_id: accessible_groups)
-            .or(Membership.where(user_id: user.id))
-            .where(identity: false)
+    query = where(user_group_id: accessible_groups, identity: false)
```

Comment:
I'll give it another try. 4.2 changed the default parenthesis rules a bit and it wasn't working right with the OR statements.